dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -594,33 +597,58 @@ private void addSecondaryVanityURLCacheCollection(final List<VanityUrl> vanityUr
     @Override
     public void validateVanityUrl(final Contentlet contentlet) {
 
-        final VanityUrl vanityUrl =
-                this.getVanityUrlFromContentlet(contentlet);
-        User user;
-        try {
-            user = this.userAPI.loadUserById(contentlet.getModUser());
-        } catch (Exception e) {
-            Logger.debug(this,e.getMessage(),e);
-            user = this.systemUser;
-        }
-        
-        final Language l = APILocator.getLanguageAPI().getLanguage(user.getLanguageId());
-        
-        if(!allowedActions.contains(vanityUrl.getAction())){
-            String message = APILocator.getLanguageAPI()
-                    .getStringKey(l, "message.vanity.url.error.invalidAction");
+        final User user = getUser(contentlet);
+        final Language language =
+                APILocator.getLanguageAPI().getLanguage(user.getLanguageId());
+
+        // check fields
+        checkMissingField(contentlet, language, VanityUrlContentType.ACTION_FIELD_VAR);
+        checkMissingField(contentlet, language, VanityUrlContentType.URI_FIELD_VAR);
+
+        final Integer action     =
+                (int)contentlet.getLongProperty(VanityUrlContentType.ACTION_FIELD_VAR);
+        final String uri         =
+                contentlet.getStringProperty(VanityUrlContentType.URI_FIELD_VAR);
+
+        if(!this.allowedActions.contains(action)){
+            String message = this.languageAPI
+                    .getStringKey(language, "message.vanity.url.error.invalidAction");
 
             throw new DotContentletValidationException(message);
         }
-      
-        if (!isValidRegex(vanityUrl.getURI())) {
-          String message = APILocator.getLanguageAPI()
-                    .getStringKey(l, "message.vanity.url.error.invalidURIPattern");
+
+        if (!isValidRegex(uri)) {
+          String message = this.languageAPI
+                    .getStringKey(language, "message.vanity.url.error.invalidURIPattern");
 
             throw new DotContentletValidationException(message);
         }
     } // validateVanityUrl.
 
+    private void checkMissingField(final Contentlet contentlet,
+                                   final Language language,
+                                   final String fieldName) {
+        if (!contentlet.getMap().containsKey(fieldName)) {
+
+            throw new DotContentletValidationException(
+                    MessageFormat.format(
```

Comment:
The error message should include the contentlet's Identifier in order to provide more information about the error.